### DISCUSSION TITLE
Difference between the various Docker Compose setups

### DISCUSSION THREAD

**Discussion by Okeanos** [SigNoz/signoz#4592] [Q&A] [answered]
So there's three distinct Docker Compose files present in the self-host demo with some subtle and some not so subtle difference:

- `docker-compose.yaml` (has lots of commented out extras, e.g. multiple zookeeper instances, and yaml-in-place-references to predefined objects)
- `docker-compose-local.yaml` (appears to build things from scratch)
- `docker-compose-core.yaml` (doesn't have a `frontend` or a `query-service` for example but some services depend on it)

The [documentation](https://signoz.io/docs/install/docker/#install-signoz-using-docker-compose) only makes a reference to `docker-compose.yaml` and at the same time there's no comment in any of the files or readme in the directory to explain what the others do.

The Makefile appears to combine `-core` and `-local` into a `run-local` command but then again `run-signoz` also uses the `--build` flag to run `docker-compose.yaml` despite not having defined any containers that need to be build locally.

Can anyone shed some light on this?

**Comment by jbalatero** [comment]
Interested on this as well

**Comment by nitishfy** [comment, marked as answer]
Hi, since the project is Open Source, the rest of compose files are mainly for contributors/our internal testing on [stagingapp.signoz.io](http://stagingapp.signoz.io/). For an end user it is recommended to use docker-compose.yaml.
